libcontainer/cgroups/fs2/memory.go

```diff
@@ -45,7 +45,12 @@ func setMemory(dirPath string, cgroup *configs.Cgroup) error {
 	if err != nil {
 		return err
 	}
-	if val := numToStr(swap); val != "" {
+	// When we set `swap`, we should let us set `memory.swap.max` to 0
+	if swap == 0 && cgroup.Resources.MemorySwap > 0 {
```

Comment:
I would change the comment to something like
```go
// memory and memorySwap set to the same value -- disable swap
```

Reply:
and rewrite the whole block to avoid mentioning `fscommon.WriteFile(dirPath, "memory.swap.max"` twice.

For example
```go
swapStr := numToStr(swap)
if swapStr == "" && swap == 0 && cgroup.Resources.MemorySwap > 0 {
  // memory and memorySwap set to the same value -- disable swap
  swapStr = "0"
}
if err := fscommon.WriteFile(dirPath, "memory.swap.max", swapStr); err != nil {
  return err
}
```